crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -0,0 +1,911 @@
+use itertools::Itertools;
+use std::cmp::Ordering;
+use std::num::NonZeroU32;
+use std::slice::Iter;
+
+use ruff_diagnostics::AlwaysFixableViolation;
+use ruff_diagnostics::Diagnostic;
+use ruff_diagnostics::Edit;
+use ruff_diagnostics::Fix;
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_codegen::Stylist;
+use ruff_python_parser::lexer::LexResult;
+use ruff_python_parser::lexer::LexicalError;
+use ruff_python_parser::Tok;
+use ruff_python_parser::TokenKind;
+use ruff_source_file::{Locator, UniversalNewlines};
+use ruff_text_size::TextRange;
+use ruff_text_size::TextSize;
+
+use crate::checkers::logical_lines::expand_indent;
+use crate::line_width::IndentWidth;
+use ruff_python_trivia::PythonWhitespace;
+
+/// Number of blank lines around top level classes and functions.
+const BLANK_LINES_TOP_LEVEL: u32 = 2;
+/// Number of blank lines around methods and nested classes and functions.
+const BLANK_LINES_METHOD_LEVEL: u32 = 1;
+
+/// ## What it does
+/// Checks for missing blank lines between methods of a class.
+///
+/// ## Why is this bad?
+/// PEP 8 recommends the use of blank lines as follows:
+/// - Two blank lines are expected between functions and classes
+/// - One blank line is expected between methods of a class.
```

Comment:
Why do we mention the recommended spacing between functions and classes if this rule is just for methods of a class? It might make sense to just say "PEP 8 recommends exactly one blank line between methods of a class". We should also not specify "missing" if this will also trigger for "too many" blank lines, instead we should say "Checks for the correct number of blank lines between methods of a class.

We could move all of this into "What it does" and have "Why is this bad" focus on consistency with PEP 8? I don't have strong feelings about that though.

Reply:
> Why do we mention the recommended spacing between functions and classes if this rule is just for methods of a class? It might make sense to just say "PEP 8 recommends exactly one blank line between methods of a class".

Done in 3d5cc8455.

> We should also not specify "missing" if this will also trigger for "too many" blank lines, instead we should say "Checks for the correct number of blank lines between methods of a class.

This rule does not trigger for too many blank lines.

Reply:
> We could move all of this into "What it does" and have "Why is this bad" focus on consistency with PEP 8? I don't have strong feelings about that though.

I don't have strong feelings about this either, I can try to change it if you want.